src/misc/health.rs

```diff
@@ -1,84 +1,125 @@
-use bevy::prelude::warn;
-use serde::Deserialize;
-
-#[derive(Deserialize, Debug, Clone)]
-pub struct Health {
-    max_health: f32,
-    health: f32,
-    #[serde(default)]
+use std::time::Duration;
+
+use bevy::prelude::*;
+use thetawave_interface::{
+    health::DamageDealtEvent,
+    spawnable::{EffectType, TextEffectType},
+};
+
+use crate::spawnable::SpawnEffectEvent;
+
+/// Handle player health regeneration
+pub fn regenerate_shields_system(mut health_query: Query<&mut HealthComponent>, time: Res<Time>) {
+    for mut health in health_query.iter_mut() {
+        health.regenerate_shields(time.delta());
+    }
+}
+
+/// Receive damage dealt events, apply damage, and spawn effects
+pub fn damage_system(
+    mut damage_dealt_events: EventReader<DamageDealtEvent>,
+    mut health_query: Query<(Entity, &Transform, &mut HealthComponent)>,
+    mut spawn_effect_event_writer: EventWriter<SpawnEffectEvent>,
+) {
+    for event in damage_dealt_events.iter() {
+        if let Ok((_entity, transform, mut health_component)) =
+            health_query.get_mut(event.target.clone())
+        {
+            // take damage from health
+            health_component.take_damage(event.damage);
+
+            // spawn damage dealt text effect
+            spawn_effect_event_writer.send(SpawnEffectEvent {
```

Comment:
text effects in a separate system. A healthPlugin that responds to events to deal with these `HealthComponent`s makes sense. E.x. These can happen in parallel and bevy might do that if they are separate systems.